dotnet/src/SemanticKernel.Abstractions/Contents/AnnotationContent.cs

```diff
@@ -42,17 +78,23 @@ public AnnotationContent()
     /// <summary>
     /// Initializes a new instance of the <see cref="AnnotationContent"/> class.
     /// </summary>
-    /// <param name="quote">The source text being referenced.</param>
+    /// <param name="label">The citation label.</param>
+    /// <param name="referenceId">Identifies the referenced resource.</param>
+    /// <param name="kind">Describes the kind of annotation</param>
     /// <param name="modelId">The model ID used to generate the content.</param>
     /// <param name="innerContent">Inner content</param>
     /// <param name="metadata">Additional metadata</param>
     public AnnotationContent(
-        string quote,
+        string label,
+        string referenceId,
+        AnnotationKind kind,
```

Comment:
- Suggest moving the kind ahead of the `referenceId`.

- I think for this class to be deserializeable you need to ad `JsonConstructor` to it, make sure the UT cover deserialization scenarios.

- I also wonder if we should have a default kind if not provided. Some annotations may also not have a reference, something like `self-referenced` annotation/link to another part of the content.

```suggestion
        AnnotationKind kind,
        string referenceId,
```

Reply:
1. Parameter order: Sure...done.
2. `JsonConstructor`: One is already defined
3. Default: I had considered a default, but it would be `None`.  Since this is only used as output/response content it will always be explicitly defined without ambiguity.